lib/ClientConnection.cc

```diff
@@ -1709,6 +1710,9 @@ void ClientConnection::checkServerError(ServerError error) {
             // https://github.com/apache/pulsar/pull/274
             closeSocket();
             break;
+        case proto::ServerError::AuthorizationError:
```

Comment:
Can you explain why do this? Because I see that the Java client has not yet dealt with it.

Reply:
When receiving the AuthenticationError or AuthorizationError, I think we should disconnect the connection. 

Right now, we only handle the AuthenticationError in Java client. See https://github.com/apache/pulsar/blob/94a6d3642800ac48b779168cbe17bb505b7c1f96/pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientCnx.java#L757

Reply:
Yes, I see. I mean, the java client also needs to be disconnected when receive AuthenticationError.